### PR TITLE
replaced hashsets by sets as mentioned in the issue

### LINKED ISSUE #3091: [good-first] Deprecate hashset
**Task description**:
After update Kubernetes dependencies to v1.26(#3080), we are able to use the [generic set](https://github.com/kubernetes/kubernetes/pull/112377). So the previous [hashset](https://github.com/karmada-io/karmada/tree/master/pkg/util/hashset) , introduced by #2958, cloud be deprecated and removed from the code base.

**Solution**:
1. Replace hashset with `sets.Set`:
```diff
diff --git a/pkg/resourceinterpreter/configurableinterpreter/configurable.go b/pkg/resourceinterpreter/configurableinterpreter/configurable.go
index 032fe7f6..131f957e 100644
--- a/pkg/resourceinterpreter/configurableinterpreter/configurable.go
+++ b/pkg/resourceinterpreter/configurableinterpreter/configurable.go
@@ -6,6 +6,7 @@ import (
        "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
        "k8s.io/apimachinery/pkg/runtime"
        "k8s.io/apimachinery/pkg/runtime/schema"
+       "k8s.io/apimachinery/pkg/util/sets"
        "k8s.io/klog/v2"

        configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
@@ -13,7 +14,6 @@ import (
        "github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/configmanager"
        "github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/luavm"
        "github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
-       "github.com/karmada-io/karmada/pkg/util/hashset"
 )

 // ConfigurableInterpreter interprets resources with resource interpreter customizations.
@@ -153,7 +153,7 @@ func (c *ConfigurableInterpreter) GetDependencies(object *unstructured.Unstructu
                return
        }

-       refs := hashset.Make[configv1alpha1.DependentObjectReference]()
+       refs := sets.New[configv1alpha1.DependentObjectReference]()
        for _, luaScript := range scripts {
                var references []configv1alpha1.DependentObjectReference
                references, err = c.luaVM.GetDependencies(object, luaScript)
@@ -164,7 +164,7 @@ func (c *ConfigurableInterpreter) GetDependencies(object *unstructured.Unstructu
                }
                refs.Insert(references...)
        }
-       dependencies = refs.List()
+       dependencies = refs.UnsortedList()

        // keep returned items in the same order between each call.
        sort.Slice(dependencies, func(i, j int) bool {

```

2. Remove [hashset ](https://github.com/karmada-io/karmada/tree/master/pkg/util/hashset) from Karmada codebase.

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by VedRatan**
**What type of PR is this?**
This PR removes the hashset and replaces it with set as mentioned in the issue.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #3091 

**Special notes for your reviewer**:
I've replaced each and every line as mentioned in the issue, carefully @RainbowMango 

**Does this PR introduce a user-facing change?**:

```release-note
NONE

```

### CODE REVIEW

File: pkg/resourceinterpreter/configurableinterpreter/configurable.go [hunk 1]
```diff
@@ -6,14 +6,15 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/klog/v2"
 
+
```
Comment (Fish-pro) [inline pkg/resourceinterpreter/configurableinterpreter/configurable.go:12]: I understand that we don't need a new row here

**Comment by XiShanYongYe-Chang** [comment]
Hi @VedRatan, you can sign off refer to https://github.com/karmada-io/karmada/pull/3103/checks?check_run_id=11067440548

File: pkg/resourceinterpreter/configurableinterpreter/configurable.go [hunk 2]
```diff
@@ -153,7 +154,7 @@ func (c *ConfigurableInterpreter) GetDependencies(object *unstructured.Unstructu
 		return
 	}
 
-	refs := hashset.Make[configv1alpha1.DependentObjectReference]()
+	refs := sets.New[configv1alpha1.DependentObjectReference]()
```
[the remark this answers is not in the document]
Comment (RainbowMango) [reply pkg/resourceinterpreter/configurableinterpreter/configurable.go:157]: +1

File: pkg/resourceinterpreter/configurableinterpreter/configurable.go (hunk 1 above)
Comment (VedRatan) [reply pkg/resourceinterpreter/configurableinterpreter/configurable.go:12]: Oh sorry, I have forgotten to signoff, will so it by today itself

Comment (VedRatan) [reply pkg/resourceinterpreter/configurableinterpreter/configurable.go:12]: Do*

**Comment by VedRatan** [comment]
> Hi @VedRatan, you can sign off refer to https://github.com/karmada-io/karmada/pull/3103/checks?check_run_id=11067440548

Sorry, I've forgotten to sign off, will do it by today itself thanks.

**Pull request closed without merging**